### PR TITLE
Unit tests for Dashboard Top Stats Cards & Dashboard Cloud Map widgets

### PR DESCRIPTION
**Pull request by victorgaard**
## Changes Made
- Added unit tests for dashboard top stats cards
- Added unit tests for dashboard cloud map widgets
- Renamed `test` for `it` in test files for better readability
- Improved dashboard cloud map component architecture
- Added dashboard cloud map widget to Storybook

## How to Test
From the `dashboard` folder
- Cloud map widget on Storybook: `npm run storybook`, then select `Komiser Widgets`
- Unit tests:  `npm run test:watch`
  - Dashboard top stats card: hit `p`, then enter `stats`
  - Dashboard cloud map: hit `p`, then enter `map`

## Screenshots
- Dashboard top stats cards
image

- Dashboard cloud map
image

- Cloud map widget on Storybook
image

### CODE REVIEW

File: dashboard/components/dashboard/components/cloud-map/DashboardCloudMap.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+const meta: Meta<typeof DashboardCloudMap> = {
+  title: 'Komiser Widgets/Cloud Map',
+  component: DashboardCloudMap,
+  tags: ['autodocs'],
+  argTypes: {
+    data: {
+      control: 'hidden'
+    }
+  },
+  decorators: [Story => <Story />]
```
Comment (eneskaya) [inline dashboard/components/dashboard/components/cloud-map/DashboardCloudMap.stories.tsx:14]: You don't really need a decorator if you're not decorating :)

File: dashboard/components/dashboard/components/top-stats/DashboardTopStats.test.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+describe('Dashboard Top Stats', () => {
+  it('should render without crashing', () => {
+    render(
+      <GlobalAppContext.Provider
+        value={{
+          ...initialContext
+        }}
+      >
+        <DashboardTopStats />
+      </GlobalAppContext.Provider>
```
Comment (eneskaya) [inline dashboard/components/dashboard/components/top-stats/DashboardTopStats.test.tsx:24]: To avoid duplicating this boilerplate, I'd recommend adding a `customRender` method as we've seen in the one example. This could look like:

```tsx
const customRender = (children: React.React.ReactElement, props: TypeOfGlobalContext) => (
  <GlobalAppContext.Provider value={{ initialContext, ...props }}>{children}</GlobalAppContext.Provider>
);
```

and then use like 

```tsx
customRender(<DashbordTopStats />, { ...some other options than initialContext })
```

File: dashboard/utils/providerHelper.test.ts [hunk 3]
```diff
@@ -1,112 +1,112 @@
 import providers from './providerHelper';
```
Comment (eneskaya) [inline dashboard/utils/providerHelper.test.ts:1]: Regarding the `providerHelper` test in general, I think it's not really adding any value. Because it's basically not testing any logic than getting something from an object by key. I'd argue for removing it altogether 👍

File: dashboard/components/dashboard/components/cloud-map/DashboardCloudMap.stories.tsx (hunk 1 above)
Comment (victorgaard) [reply dashboard/components/dashboard/components/cloud-map/DashboardCloudMap.stories.tsx:14]: oops, copy paste residuals 😅

File: dashboard/components/dashboard/components/top-stats/DashboardTopStats.test.tsx (hunk 2 above)
Comment (victorgaard) [reply dashboard/components/dashboard/components/top-stats/DashboardTopStats.test.tsx:24]: Nice one!

Comment (victorgaard) [reply dashboard/components/dashboard/components/top-stats/DashboardTopStats.test.tsx]: I also improved the test a little bit by mocking the data and seeing the component render when there's data.

**Comment by victorgaard** [comment]
Had to rebase with develop and do an adjustment. Marking it as ready for review again.

File: dashboard/components/input/Input.tsx [hunk 4]
```diff
@@ -43,7 +43,7 @@ function Input({
     const trimmedValue = e.target.value.trim();
     if (!regex || !trimmedValue) return;
 
-    const testResult = regex.test(trimmedValue);
+    const testResult = regex.it(trimmedValue);
```
Comment (eneskaya) [inline dashboard/components/input/Input.tsx:46]: is this change intended? 🤔

Comment (victorgaard) [reply dashboard/components/input/Input.tsx:46]: It was definitely not! Great catch :)

Comment (victorgaard) [reply dashboard/components/input/Input.tsx]: Good reminder to write tests for the input lol

**Review by eneskaya — APPROVED** [review]
